src/struc/model.jl

```diff
@@ -0,0 +1,826 @@
+
+"""`Array` of protein backbone atom names."""
+const backbone_atom_names = AbstractString["CA", "N", "C"]
+
+"""Determines if an `AbstractAtom` is not a hetero-atom and corresponds to a
+protein backbone atom."""
+backboneselector(atom::AbstractAtom) = stdatomselector(atom) && atomnameselector(atom, backbone_atom_names)
+
+"""`Array` of protein standard heavy atom names."""
+const heavy_atom_names = AbstractString["C", "CA", "CB", "CD", "CD1", "CD2",
+                    "CE", "CE1", "CE2", "CE3", "CG", "CG1", "CG2", "CH2", "CZ",
+                    "CZ2", "CZ3", "N", "ND1", "ND2", "NE", "NE1", "NE2", "NH1",
+                    "NH2", "NZ", "O", "OD1", "OD2", "OE1", "OE2", "OG", "OG1",
+                    "OH", "SG"]
+
+"""Determines if an `AbstractAtom` is not a hetero-atom and corresponds to a
+standard protein heavy atom."""
+heavyatomselector(atom::AbstractAtom) = stdatomselector(atom) && atomnameselector(atom, heavy_atom_names)
+
+"""Determines if an `AbstractResidue` or `AbstractAtom` has its resiudue name in
+the given `Array`."""
+resnameselector(element::Union{AbstractResidue, AbstractAtom}, res_names::Array{AbstractString,1}) = getresname(element) in res_names
+
+# Any more?
+"""`Array` of residue names corresponding to water."""
+const water_res_names = AbstractString["HOH"]
+
+"""Determines if an `AbstractResidue` or `AbstractAtom` represents a water
+molecule."""
+waterselector(element::Union{AbstractResidue, AbstractAtom}) = resnameselector(element, water_res_names)
+
+"""Determines if an `AbstractResidue` represents a standard protein residue,
+i.e. consists of ATOM records."""
+stdresselector(res::AbstractResidue) = !ishetres(res)
+
+"""Determines if an `AbstractResidue` represents a hetero molecule, i.e.
+consists of HETATM records."""
+hetresselector(res::AbstractResidue) = ishetres(res)
+
+"""Determines whether an `AbstractAtom` or `AbstractResidue` is disordered, i.e.
+has multiple locations in the case of atoms or multiple residue names (point
+mutants) in the case of residues."""
+disorderselector(atom::AbstractAtom) = typeof(atom) == DisorderedAtom
+disorderselector(res::AbstractResidue) = typeof(res) == DisorderedResidue
+
+# Either the element is H or the element field is empty, the atom name contains an H and there are no letters in the atom name before H
+# For example atom names 1H and H1 would be hydrogens but NH1 would not
+"""Determines if an `AbstractAtom` represents hydrogen. Uses the element field
+where possible, otherwise uses the atom name."""
+hydrogenselector(atom::AbstractAtom) = getelement(atom) == "H" || (getelement(atom) == "" && 'H' in getatomname(atom) && !ismatch(r"[a-zA-Z]", getatomname(atom)[:findfirst(getatomname(atom), 'H')-1]))
+
+
+function show(io::IO, struc::Structure)
+    model = struc[1]
+    println(io, "Name                 -  ", getstrucname(struc))
+    println(io, "No models            -  ", countmodels(struc))
```

Comment:
I agree it is ambiguous as stands - will implement a change. Perhaps it is okay to omit N/number entirely? e.g.

```
Name              -  1ABC.pdb
Models            -  1
Chain(s)          -  AB
Residues          -  500
Point mutations   -  4
Other molecules   -  2
```
etc.